### ISSUE TITLE
shadow_unrelated triggers on destructuring

### ISSUE THREAD

**Issue by purplesyringa** [rust-lang/rust-clippy#13795]
### Summary

```rust
let Struct { value, .. } = value;
```
triggers `shadow_unrelated`.

### Lint Name

shadow_unrelated

### Reproducer

I tried this code:

```rust
#![warn(clippy::shadow_unrelated)]

struct Struct {
    value: i32,
}

fn f(value: Struct) {
    let Struct { value, .. } = value;
}
```

I saw this happen:

```
warning: `value` shadows a previous, unrelated binding
 --> src/lib.rs:8:18
  |
8 |     let Struct { value, .. } = value;
  |                  ^^^^^
  |
note: previous binding is here
 --> src/lib.rs:7:6
  |
7 | fn f(value: Struct) {
  |      ^^^^^
```

I expected to see this happen:

No warning.

### Version

```text
rustc 1.84.0-nightly (1e4f10ba6 2024-10-29)
binary: rustc
commit-hash: 1e4f10ba6476e48a42a79b9f846a2d9366525b9e
commit-date: 2024-10-29
host: x86_64-unknown-linux-gnu
release: 1.84.0-nightly
LLVM version: 19.1.1
```
